decimalSuite_test.go

```diff
@@ -55,73 +55,74 @@ func TestFromSuite(t *testing.T) {
-	// split into groups: testName, TestFunct, TestVals (x2) and TestAns
-	regex := `(\n)(?P<TestName>dd[\w]*)(\s*)(?P<TestFunc>[\S]*)(\s*\'?)(?P<TestVals_1>(\+|-)*[^\t\f\v\' ]*)('?\s*'?)(?P<TestVals_2>(\+|-[^->])?[^\t\f\v\' ]*)('?\s*->\s*'?)(?P<answer>(\+|-)*[^\r\n\t\f\v\' ]*)`
-	r := regexp.MustCompile(regex)
+	r := regexp.MustCompile((`(?:\n)` + // start with newline (?: non capturing group)
+		`(?P<TestName>dd[\w]*)` + // first capturing group: testfunc made of anything that isn't a whitespace
+		`(?:\s*)` + // match any whitespace (?: non capturing group)
+		`(?P<TestFunc>[\S]*)` + // testfunc made of anything that isn't a whitespace
+		`(?:\s*\'?)` + // after can be any number of spaces and quotations if they exist (?: non capturing group)
+		`(?P<TestVals_1>\+?-?[^\t\f\v\' ]*)` + // first test val is anything that isnt a whitespace or a quoteation mark
+		`(?:'?\s*'?)` + // match any quotation marks and any space (?: non capturing group)
+		`(?P<TestVals_2>\+?-?[^->]?[^\t\f\v\' ]*)` + //testvals2 same as 1 but specifically dont match with '->'
+		`(?:'?\s*->\s*'?)` + // matches the indicator to answer and surrounding whitespaces (?: non capturing group)
+		`(?P<answer>\+?-?[^\r\n\t\f\v\' ]*)`)) // matches the answer that's anything that is plus minus but not quotations
+	// capturing gorups are testName, TestFunct, TestVals_1,  TestVals_2, and answer)
+
 	ans := r.FindAllStringSubmatch(dataString, -1)
-	var datum testCaseStrings
-	for i := range ans {
-		datum.op = ans[i][4]
-		datum.val1 = ans[i][6]
-		datum.val2 = ans[i][9]
-		datum.expectedResult = ans[i][12]
-		datum.name = ans[i][2]
-		data = append(data, datum)
+	for _, a := range ans {
+		data = append(data, testCaseStrings{
+			name:           a[1],
+			op:             a[2],
+			val1:           a[3],
+			val2:           a[4],
+			expectedResult: a[5]})
 
 	}
-	return data
+	return
 
 }
 
-//convertToDec64 converts the map object strings to decimal64s
+// convertToDec64 converts the map object strings to decimal64s
 func convertToDec64(testvals testCaseStrings) (dec64Vals decValContainer) {
-	dec64Vals.val1, dec64Vals.val1Err = ParseDecimal64(testvals.val1)
-	dec64Vals.expectedResult, dec64Vals.expecValErr = ParseDecimal64(testvals.expectedResult)
-	dec64Vals.val2, dec64Vals.val2Err = ParseDecimal64(testvals.val2)
+	dec64Vals.val1.d, dec64Vals.val1.err = ParseDecimal64(testvals.val1)
+	dec64Vals.expected.d, dec64Vals.expected.err = ParseDecimal64(testvals.expectedResult)
+	dec64Vals.val2.d, dec64Vals.val2.err = ParseDecimal64(testvals.val2)
 	return
 }
 
-// TODO: get doTests to run more functions
-//doTests completes the tests and returns a boolean and string on if the test passes
-func doTests(testVals decValContainer, testValStrings testCaseStrings) (testStatus bool, testString string) {
+// doTest completes the tests and returns a boolean and string on if the test passes
+func doTest(testVals decValContainer, testValStrings testCaseStrings) (testFailed bool, testString string) {
```

Comment:
tl;dr: `testPassed` is better.

Avoid Booleans with a "negative" connotation. `testFailed` is, strictly speaking, a positive assertion (a failure _did_ occur). But psychologically it carries the connotation that a desired outcome _didn't_ occur.

Just to be clear, this isn't to make programmers feel better through the power of positive thinking. It's simply to avoid the caller having to think in double negatives.

Reply:
does this give me permission to use ```if testPassed == false ``` or ```if !testPassed``` 
I don't want to do anything if the test passed, therefore I don't need to use ```if testPassed ``` 

see original comment:

> Booleans should be named as if they are logical propositions (statements of fact). This is because they represent the truth-or-falsity of a statement. It also makes tests read more naturally and reduces the temptation to needlessly add `== true` or `== false` to such tests.
> 
> In this case, `testPassed` would be a good choice (`if testPassed …`).

Reply:
`if !testPassed`. If you use `if testPassed == false`, you won't get invited to parties.